### ISSUE TITLE
mpv will not render video after DPMS is turned off/on on vulkan

### ISSUE THREAD

**Issue by LaserEyess** [mpv-player/mpv#9016]
### Important Information

Provide following Information:
- mpv version: git master
- Linux Distribution and Version: Arch Linux
- Source of the mpv binary: compiled from source
- If known which version of mpv introduced the problem: c26d83348b730d083f3be313556039fd31d32505
- Window Manager and version: `sway version 1.6-71140301 (Jul  8 2021, branch 'master')`
- GPU driver and version: AMD Navi, using amdgpu on 5.13.4, RADV from mesa 21.1.5 
- Possible screenshot or video of visual glitches: N/A

### Reproduction steps
1. `mpv --no-config --gpu-api=vulkan --gpu-context=waylandvk file.mkv`, any file will do
2. Pause the video and wait until DPMS gets turned off due to idle activity
3. wait around 15 minutes, or some sufficient amount of time for sway to completely stop sending frame callbacks (?)
4. unlock the screen/wake up monitors and unpause mpv
5. rendering will have completely stopped, an unscaled frame will be present on the screen. Audio and input is unaffected but there's no OSD

### Expected behavior
mpv will resume properly and render video again

### Actual behavior
mpv does not render video

### Log file
https://0x0.st/-VTh.log

wayland client debug: https://0x0.st/-VAo.log

### Sample files
Any file should do

Bisected to c26d83348b730d083f3be313556039fd31d32505. This is *potentially* a bad bisect

**Comment by Dudemanguy** [comment]
This is by no means a proper fix but does the issue still happen with this really dumb patch?

```
diff --git a/video/out/wayland_common.c b/video/out/wayland_common.c
index 4157782c0c..eee6445b42 100644
--- a/video/out/wayland_common.c
+++ b/video/out/wayland_common.c
@@ -1540,8 +1540,10 @@ static void vo_wayland_dispatch_events(struct vo_wayland_state *wl, int nfds, in
     if (fds[0].revents & POLLIN)
         wl_display_dispatch_pending(wl->display);
 
-    if (fds[1].revents & POLLIN)
+    if (fds[1].revents & POLLIN) {
         mp_flush_wakeup_pipe(wl->wakeup_pipe[0]);
+        wl->pending_vo_events |= VO_EVENT_EXPOSE;
+    }
 }
 
 /* Non-static */
```

**Comment by LaserEyess** [comment]
The above snippet does fix the issue, though I have not tested #9023 yet. In the meantime I've discovered how to reproduce this bug 100% and the minimum viable settings, I'll update the OP with it but basically it's:
```
mpv --no-config --gpu-api=vulkan --gpu-context=waylandvk file.mkv
```

Any file will do. I've bisected this behavior to c26d83348b730d083f3be313556039fd31d32505. This could be a bad bisect but I am able to trigger the bug consistently, even if it takes around 15-30 minutes to trigger.
